clab/config/template.go

```diff
@@ -43,21 +47,15 @@ func LoadTemplates(tmpl *template.Template, role string) error {
 }
 
 func RenderAll(nodes map[string]nodes.Node, links map[int]*types.Link) (map[string]*NodeConfig, error) {
+	// A map with the ShortName as the key
 	res := make(map[string]*NodeConfig)
 
-	if len(TemplatePaths) == 0 {
-		return nil, fmt.Errorf("please specify one of more paths with --template-path")
+	if len(TemplatePaths) == 0 { // default is the install path
+		TemplatePaths = []string{"@"}
 	}
 
 	if len(TemplateNames) == 0 {
-		var err error
-		TemplateNames, err = GetTemplateNamesInDirs(TemplatePaths)
```

Comment:
Thinking about this again, I'm not sure this is the best approach: 
- template paths contain many templates
- templates can be for different roles, so you might miss completely
- templates are supposed to be applied in order

so auto-discovery from file will not be reliable/cause confusion

Reply:
the idea for `GetTemplateNamesInDirs` is only to kick in if the template names were not supplied by a user

I see this to be a common use cases:
1. you either have just one template in a folder that you want to execute. In that case your template path points to a certain dir and in this dir you have a single template file. No need to ask a user to specify this template name, it will be discovered
2. you have many templates which are stacked and represent a step-by-step configuration process. Since the templates are stacked, I see it is convenient to name them in the order so that you see which one follows the other. For example: 010-ifaces, 020-igp, 030-bgp  
  this will allow the function to follow the order that is assumed by a template designer

Reply:
👍 ok, the second use case is indeed something that should be very useful. 

For the base templates it could result in a great amount of things being deployed in no order, but luckily its quick to revert anything using config (just start fresh)

Reply:
added this back and updated the function not to include duplicates or blanks